### ISSUE TITLE
Chapter 07. Decompiling using the ILSpy extension for Visual Studio Code

### ISSUE THREAD

**Issue by Dreamoochy** [markjprice/cs10dotnet6#86]
1. I don't have two tabs with C# and IL. It is possible to split editor and get two tabs of the decompiled assembly, but they are synchronized. I.e. if I change lang view from C# to IL, both views are switched to IL.
2. There's no `.module extern libSystem.Native` text in the decompiled IL.
3. Decompiled assembly tree has way less nodes:
`````csharp
v System.IO.FileSystem
  v {}
    v <Module>
`````
So it is impossible to perform steps 9, 10, 11.

Maybe it's because my ilspy-vscode extension is v0.13.1 while the one in the book is 0.9.0. Something might have changed. And/or some additional configuring is required.

I work with Win 11 21H2, VSCodium v1.68.1, ilspy-vscode v0.13.1, .NET SDK v6.0.301

**Comment by markjprice** [comment]
Yes, it looks like that extension has been changed. You could go to the extension and click Uninstall | Install Another Version... and select 0.11.2 which is the newest version that shows two edit windows side-by-side as shown in the book. Then you can upgrade back again.

![image](https://user-images.githubusercontent.com/14040265/175518546-079f3543-f6b5-43bf-9a76-38187d437cc5.png)

![image](https://user-images.githubusercontent.com/14040265/175518644-cdcac527-3dd2-4251-80c2-5a6e9241dac7.png)

**Comment by markjprice** [comment]
In the next edition, I will tell the reader to open a different assembly so it works with the latest version of ILSpy. Probably the `System.Linq` assembly and decompile the `Enumerable` class and its `Average` method for a sequence of `int` values:

![image](https://user-images.githubusercontent.com/14040265/175520803-4f3e0442-f91d-436a-8f74-925e48937da4.png)

And then show how to toggle between C# and IL code:

![image](https://user-images.githubusercontent.com/14040265/175520894-2a8e3987-cfec-4a4a-81bb-19d211c901c2.png)

**Comment by markjprice** [comment]
I have added an errata item:
https://github.com/markjprice/cs10dotnet6/blob/main/errata.md#page-297---decompiling-using-the-ilspy-extension-for-visual-studio-code
